### PR TITLE
Feature/source jump

### PR DESCRIPTION
**Pull request by fscholdei**
## Proposed changes

Add link to source file in diff sections

### Your checklist for this pull request

- [x] PR is well described
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [x] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by sdorra** [comment]
Blocked until 2.3.1 is out.

File: scm-ui/ui-components/src/repos/DiffFile.tsx [hunk 1]
```diff
@@ -411,29 +415,31 @@ class DiffFile extends React.Component<Props, State> {
     }
     const collapseIcon = this.hasContent(file) ? <Icon name={icon} color="inherit" /> : null;
     const fileControls = fileControlFactory ? fileControlFactory(file, this.setCollapse) : null;
-    const sideBySideToggle =
-      file.hunks && file.hunks.length > 0 ? (
```
Comment (eheimbuch) [inline scm-ui/ui-components/src/repos/DiffFile.tsx:415]: Why did you remove the check if the file length is greater than 0?

File: scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx [hunk 2]
```diff
@@ -153,6 +154,38 @@ class RepositoryRoot extends React.Component<Props> {
       redirectedUrl = url + "/info";
     }
 
+    const fileControlFactoryFactory: (changeset: Changeset) => FileControlFactory = changeset => file => {
```
Comment (eheimbuch) [inline scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx:157]: This component is already way to big. You should move this function to a separate file.
